v2/apiserver/internal/core/mongodb/events_store.go

```diff
@@ -260,7 +261,7 @@ func (e *eventsStore) CancelMany(
 
 	// Bail if we're not canceling pending, starting, or running events
 	if !cancelPending && !cancelStarting && !cancelRunning {
-		return events, nil
+		return eventCh, errCh, nil
```

Comment:
Nothing will ever close these channels, so whoever's reading from these channels will get stuck forever.

You probably want to return nils here.